packages/functional-tests/src/functional-tests.local.spec.ts

```diff
@@ -0,0 +1,159 @@
+            a11yServiceClient = new A11yServiceClient(cred, `https://apim-${apimName}.azure-api.net`);
+            onDemandPageScanRunResultProvider = container.get(OnDemandPageScanRunResultProvider);
+            guidGenerator = container.get(GuidGenerator);
+            logger = container.get(Logger);
+            testRunner = container.get(TestRunner);
+            await logger.setup({
+                source: 'dev box',
+            });
+            testRunner.setLogger(logger);
+            testContextData = {
+                scanUrl: 'https://www.washington.edu/accesscomputing/AU/before.html',
+            };
+            releaseIdStub = `dev-${guidGenerator.createGuid()}`;
+            runIdStub = `dev-${guidGenerator.createGuid()}`;
+        } else {
+            console.log('One or more service credentials is missing. Skipping All functional tests.');
+        }
+    });
+
+    testIf('PostScan', isServiceCredProvided, async (done: jest.DoneCallback) => {
+        const postScanTests = getTests('PostScan');
+        await testRunner.run(postScanTests, TestEnvironment.all, releaseIdStub, runIdStub);
+        done();
+    });
+
+    testIf('ScanStatus', isServiceCredProvided, async (done: jest.DoneCallback) => {
+        const scanStatusTests = getTests('ScanStatus');
+        testContextData.scanId = (await a11yServiceClient.postScanUrl(testContextData.scanUrl)).body[0].scanId;
+        await testRunner.run(scanStatusTests, TestEnvironment.all, releaseIdStub, runIdStub);
+        done();
+    });
+
+    testIf('ScanExecution', isServiceCredProvided, async (done: jest.DoneCallback) => {
+        const scanRunResultResponse = await waitForScanRequestCompletion();
+        expect(scanRunResultResponse.reports.length).toEqual(2);
+        testContextData.reportId = scanRunResultResponse.reports[0].reportId;
+        done();
+    });
+
+    testIf('ScanPreProcessing', isServiceCredProvided, async (done: jest.DoneCallback) => {
+        const scanPreProcessingTests = getTests('ScanPreProcessing');
+        await testRunner.run(scanPreProcessingTests, TestEnvironment.all, releaseIdStub, runIdStub);
+        done();
+    });
+
+    testIf('ScanQueueing', isServiceCredProvided, async (done: jest.DoneCallback) => {
+        const scanQueueingTests = getTests('ScanQueueing');
+        await testRunner.run(scanQueueingTests, TestEnvironment.all, releaseIdStub, runIdStub);
+        done();
+    });
+
+    testIf('ScanReports', isServiceCredProvided, async (done: jest.DoneCallback) => {
+        const scanReportsTests = getTests('ScanReports');
+        await testRunner.run(scanReportsTests, TestEnvironment.all, releaseIdStub, runIdStub);
+        done();
+    });
+
+    testIf('Finalizer', isServiceCredProvided, async (done: jest.DoneCallback) => {
+        const finalizerTests = getTests('Finalizer');
+        await testRunner.run(finalizerTests, TestEnvironment.all, releaseIdStub, runIdStub);
+        done();
+    });
```

Comment:
It looks very similar implementation. Can we do some generic function that get list of tests groups to run in a sequence?

Reply:
yep. updated.